### PR TITLE
add attribute resize tracking

### PR DESCRIPTION
**Pull request by aardgoose**
Untangle overloading of BufferAttribute.dynamic property, re #14730 

.dynamic sets buffer usage expectations as documented.

.updateRange works as documented if dynamic = false. (currently updates entire buffer)

.setArray() and .copy() are safe to use if dynamic = true. (currently produces error)

By default gl.bufferSubData is used for updates except in resize operations.

(a flag to select update mechanism could be added if required )

### CODE REVIEW

**Comment by Mugen87** [comment]
I'm a bit irritated now. I always thought `three.js` does not allow to resize buffer attributes. If an app needs a bigger attribute buffer for some reasons, it has to create a new attribute. Or create a sufficient sized buffer in the forefront.

**Comment by aardgoose** [comment]
Some developers (not myself) were resizing them it seems, which is what started the initial discussion referred to in #14730.  

After the changes I added which allowed you to drop the arrays from the JS side, after the GPU had been updated ( via the onUpload() callback) - this user 'requirement' came out the woodwork resulted in dynamic flag overloading.

File: src/core/BufferAttribute.js [hunk 1]
```diff
@@ -24,6 +24,7 @@ function BufferAttribute( array, itemSize, normalized ) {
 
 	this.dynamic = false;
 	this.updateRange = { offset: 0, count: - 1 };
+	this.resized = false;
```
Comment (WestLangley) [inline src/core/BufferAttribute.js]: I think `.resized` should be private... somehow...

**Comment by Mugen87** [comment]
Does the following error message still make sense?

https://github.com/mrdoob/three.js/pull/15798/files#diff-54f00f0a09b102e7d4bffbf907becd3aR89

File: src/core/BufferAttribute.js (hunk 1 above)
Comment (Mugen87) [reply src/core/BufferAttribute.js]: Maybe `._resized`?

Comment (aardgoose) [reply src/core/BufferAttribute.js]: is there a convention to indicate private properties? 
_resized looks reasonable.

Comment (WestLangley) [reply src/core/BufferAttribute.js]: Yes, prefixing with underscore has been a convention used in this library.

Comment (aardgoose) [reply src/core/BufferAttribute.js]: ok.will do.

**Comment by aardgoose** [comment]
Nothing bad appears to happen passing an empty TypedArray in the count = 0 error case.
Tested on Chrome/FF/IE11 on Windows 10, so maybe a pointless error other than pointing to programming error elsewhere.

**Pull request closed without merging**